### PR TITLE
[8.0][IMP] Improve possibility of inheritance for values generation of writeoff move lines

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Add a method to perpare values for writeoff move lines to improve possibility of inheritance.

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
Any news about this issue ?

**Comment by adrienpeiffer** [comment]
Any news about this PR ?

**Comment by adrienpeiffer** [comment]
OPW 645038

File: addons/account/account_move_line.py [hunk 1]
```diff
@@ -917,6 +917,24 @@ def reconcile_partial(self, cr, uid, ids, type='auto', context=None, writeoff_ac
...
+        return {'name': name,
+                'debit': debit,
+                'credit': credit,
+                'account_id': account_id,
+                'date': date,
+                'analytic_account_id': analytic_account_id,
+                'partner_id': partner_id,
+                'currency_id': cur_id or (account_currency_id or False),
+                'amount_currency': amount_currency and -1 * amount_currency or
+                (account_currency_id and -1 * currency or 0.0)
```
Comment (nim-odoo) [inline addons/account/account_move_line.py]: This doesn't seem correct: multiplying by -1 should not be done for writeoff_acc_id

**Comment by nim-odoo** [comment]
Beside the remark I made, I am not a big fan of sending so many parameters in the method. This makes the understanding of the method quite complex. For example, you send `cur_id`, `currency` and `account_currency_id`: inside the method, it is simply impossible to understand to what these various currencies correspond to.

Maybe a clearer way would be to use the method simply to create the dictionary, i.e. the method would simply take as parameters the final value of every dictionary key.

I would need the opinion of @beledouxdenis

**Comment by adrienpeiffer** [comment]
Hi @nim-odoo,
Thanks for your feedback !
I modified this one according your comment.

**Comment by beledouxdenis** [comment]
Hello Adrien,

These kinds of pull requests in stable releases are exceptionally accepted, if the use case worth it and there is indeed no other way to perform the override.

As you do not explain your use case, there is currently no way for us to know if your use case worth the effort (or the risks).

Can you therefore explain your use case more specifically, why you need to add or change values within this specific dict, so we can evaluate if your pull request brings an added value to the product.

Regards,

**Comment by adrienpeiffer** [comment]
Hi Denis !

Thanks for your answer.
In our case, we need to put additional information in the data of writeoff's journal items . These data are selected from the reconciliation wizard.

Regards,

Adrien

**Comment by beledouxdenis** [comment]
Hello Adrien, 

Can you please be more specific than
>put additional information

What kind of data ? Is it legal data ? Accounting data ?

Regards,

**Comment by adrienpeiffer** [comment]
>What kind of data ? Is it legal data ? Accounting data ?

It's some custom analytic axis

**Comment by beledouxdenis** [comment]
I am sorry but that is still not enough specific for us to allow this pull request.

**Comment by adrienpeiffer** [comment]
What do you mean by not enough specific ? This is only one situation in which there is a problem with that.
Regardless, I find that the preparation of the values in separate methods is a good thing

**Comment by beledouxdenis** [comment]
> I find that the preparation of the values in separate methods is a good thing

Indeed, but, for a stable release such as 7.0 and 8.0, we accept those only if the use case requires it and worth it. 

I would have no problem to merge this in master.

Therefore, if you do not explain exactly why you need this, what are the requirements that make this change in a stable release extremely necessary, we won't merge it.

**Comment by adrienpeiffer** [comment]
It's just two more analytical axes that can be selected from the reconciliation wizard. these axes are present as columns in the journal items. Therefore, at the time of creation, we need to fill these key values in the dictionary.

We understand you may not be able to merge it on 8.0.

**Comment by beledouxdenis** [comment]
I will say it otherwise: For what feature do you need this pull request ? What is your customer asking you to do, with customer words (not with developer words) that make this pull request mandatory ?

**Comment by sbidoul** [comment]
@beledouxdenis the use case goes as follow.

Instead of one analytic axis the customer needs 3 (project, department, financing source to be precise). Each account.move.line must be allocated to the 3 analytic axis. Analytic distributions do not fit the requirement (for usability and reporting reasons), therefore we need to add two new columns on account.move.line. 

Without the refactoring proposed in this PR we cannot let the user select the two other analytic axis when doing writeoffs in reconciliations.

As a side note, this requirement of adding multiple classification axis on account.move.line is a common one that I have seen several times on the mailing list too. So this might be a useful feature to consider in the future.

**Comment by sbidoul** [comment]
@beledouxdenis is my explanation clear? Can you unblock this PR?

**Comment by adrienpeiffer** [comment]
@beledouxdenis have you considered @sbidoul explanation ?

**Comment by mart-e** [comment]
Hello,

Thank you for your contribution but the version 8.0 is no longer supported. We only support the last 3 stable versions so no longer accepts patches into this branch.
We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.
Regards

**Pull request closed without merging**